### PR TITLE
MAUI-2438 DataGrid selection UG

### PR DESCRIPTION
**Pull request by FarjanaParveen**
DataGrid selection UG

### CODE REVIEW

File: MAUI/DataGrid/selection.md [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+<th> Modes </th>
+<th> Description </th>
+</tr>
+<tr>
+<td> {{'[None] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionMode.html#Syncfusion_Maui_DataGrid_DataGridSelectionMode_None)'| markdownify }} </td>
+<td>Disables selection, and no rows or cells can be selected. This is the default value.</td>
+</tr>
+<tr>
+<td>{{'[Single] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionMode.html#Syncfusion_Maui_DataGrid_DataGridSelectionMode_Single)'| markdownify }} </td>
+<td> Allows selection of a single row or cell only. Upon selecting the next row or cell, the selection in the previous row or cell is cleared. </td>
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:29]: @FarjanaParveen  - remove the **"cell only"** content.

File: MAUI/DataGrid/selection.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+</tr>
+<tr>
+<td>{{'[Multiple] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionMode.html#Syncfusion_Maui_DataGrid_DataGridSelectionMode_Multiple)'| markdownify }}</td>
+<td> Allows selection of more than one row or cell. Selection is not cleared when selecting more than one row or cell. When you click on an already selected row or cell for the second time, the selection is cleared. </td>
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:33]: @FarjanaParveen  - Remove the "or cell" content.

File: MAUI/DataGrid/selection.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+<td> Allows selection of a single row or cell only. However, upon tapping the row or cell again, the selection is cleared. Similar to single mode, upon selecting the next row or cell, the selection in the previous row or cell is cleared. </td>
+</tr>
+</table>
+
+## Disable selection for rows and columns
+You can disable selection in a particular row or cell or column by handling the [CurrentCellActivating] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentCellActivating) event.
+
+## Getting selected rows
+`SfDataGrid` provides [SelectedIndex] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SelectedIndex), [SelectedRow] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SelectedRow), and [CurrentRow] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentRow) properties to get details of the selected rows when the selection mode is Single, Multiple, and SingleDeselect.
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:45]: @FarjanaParveen  - Include the SelectedRows property too in this para.

File: MAUI/DataGrid/selection.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+
+`SfDataGrid.SelectedRow`: Provides the underlying data object of the selected row. Denotes the underlying data object of the first selected row in multiple selection.
+
+`SfDataGrid.SelectedIndex`: Provides the row index of SfDataGrid.SelectedItem. Denotes the index of first selected row in multiple selection.
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:49]: SfDataGrid.SelectedItem - This is wrong API right?

File: MAUI/DataGrid/selection.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+        dataGrid.SelectedRows.Add(viewModel.OrderInfoCollection[11]);
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+![Multiple Selection in .NET MAUI DataGrid](Images/selection/maui-datagrid-multiple-selection.png)
+
+### Process selection using methods
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:104]: @FarjanaParveen  - Change the title as,

Select all the rows

File: MAUI/DataGrid/selection.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+    public MainPage()
+    {
+        InitializeComponent();
+        dataGrid.SelectAll();
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![SelectionAll in .NET MAUI DataGrid](Images/selection/maui-datagrid-selectall.png)
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:119]: Alt text may be

All the rows selected in .NET MAUI DataGrid

File: MAUI/DataGrid/selection.md [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+
+        //Clear selection using clear selection method.
+        dataGrid.ClearSelection();
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Events in selection
+The data grid provides the following events for selection:
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:300]: @FarjanaParveen  - Don't split datagrid word.

File: MAUI/DataGrid/selection.md [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+
+[SelectionChanging] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SelectionChanging): This event is raised while selecting a row at the execution time before the row is selected. So it allows canceling the selection action by setting the Cancel property of DataGridSelectionChangingEventArgs.
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:302]: @FarjanaParveen  - Put the api reference link for Cancel and DataGridSelectionChangingEventArgs.

File: MAUI/DataGrid/selection.md [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+[SelectionChanged] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortColumnsChanged): This event is raised after a row is selected.
+These two events are triggered with DataGridSelectionChangingEventArgs and DataGridSelectionChangedEventArgs that contain the following properties:
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:304]: @FarjanaParveen  - Put the backtik for APIs.

File: MAUI/DataGrid/selection.md [hunk 10]
```diff
@@ -0,0 +1,468 @@
...
+{% highlight xaml tabtitle="MainPage.xaml"%}
+<ContentPage xmlns:syncfusion="http://schemas.syncfusion.com/maui">
+    <ContentPage.Content>
+        <syncfusion:SfDataGrid ItemsSource="{Binding OrderInfoCollection}"
+                            SelectedIndex="{Binding DataGridSelectedIndex}"
+                            SelectedRow="{Binding DataGridSelectedRow}"/>
+    </ContentPage.Content>
+</ContentPage>
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
```
Comment (NeelakandanKannan) [inline MAUI/DataGrid/selection.md:427]: @FarjanaParveen  = file name should be ViewModel.cs. Provide the full class implementation.

**Review by NeelakandanKannan — CHANGES_REQUESTED** [review]
I have added my feedback as comments. Get back to me with the changes.

File: MAUI/DataGrid/selection.md [hunk 11]
```diff
@@ -0,0 +1,547 @@
...
+<td>
+Moves the current cell to previous to the active current cell. If the CurrentCell is in first cell, pressing <kbd>LeftArrow</kbd> does nothing. 
+</td>
+</tr>
+<tr>
+<td>
+<kbd>RightArrow</kbd>
+</td>
+<td>
+Moves the current cell to next to the active current cell. If the CurrentCell is in last cell, pressing <kbd>RightArrow</kbd> does nothing.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:159]: Moves the current cell next to the active current cell. If the CurrentCell is in the last cell, pressing the <kbd>RightArrow</kbd> does nothing.

File: MAUI/DataGrid/selection.md [hunk 12]
```diff
@@ -0,0 +1,547 @@
...
+<td>
+Moves the current cell to the last cell of the current row.
+</td>
+</tr>
+<tr>
+<td>
+<kbd>PageDown</kbd>
+</td>
+<td>
+The SfDataGrid will be scrolled to next set of rows that are not displayed in view, including the row that are partially displayed and the current cell is set to last row.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:183]: The SfDataGrid will be scrolled to the next set of rows that are not displayed in the view, including the row that is partially displayed and the current cell is set to the last row.

File: MAUI/DataGrid/selection.md [hunk 13]
```diff
@@ -0,0 +1,547 @@
...
+</td>
+</tr>
+<tr>
+<td>
+<kbd>PageUp</kbd>
+</td>
+<td>
+The SfDataGrid will be scrolled to previous set of rows that are not displayed in view, including the row that are partially displayed and the current cell is set to the first row.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:191]: The SfDataGrid will be scrolled to the previous set of rows that are not displayed in the view, including the row that is partially displayed and the current cell is set to the first row.

File: MAUI/DataGrid/selection.md [hunk 14]
```diff
@@ -0,0 +1,547 @@
...
+</td>
+</tr>
+<tr>
+<td>
+<kbd>Tab</kbd>
+</td>
+<td>
+Moves the current cell to next to the active current cell. If the active current cell is the last cell of the current row, the focus will moved to first cell of the row next to the current row.If the active current cell is the last cell of the last row, the focus will be moved to next control in the tab order of the parent container.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:199]: Moves the current cell next to the active current cell. If the active current cell is the last cell of the current row, the focus will be moved to the first cell of the row next to the current row. If the active current cell is the last cell of the last row, the focus will be moved to the next control in the tab order of the parent container.

File: MAUI/DataGrid/selection.md [hunk 15]
```diff
@@ -0,0 +1,547 @@
...
+</td>
+</tr>
+<tr>
+<td>
+<kbd>Shift</kbd> + <kbd>Tab</kbd>
+</td>
+<td>
+Moves the current cell to previous to the active current cell. If the active current cell is the first cell of the current row, the current cell will moved to last cell of the row previous to the current row.If the active current cell is the first cell of the first row, the focus will be moved to previous control in the tab order of the parent container.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:207]: Moves the current cell to previous to the active current cell. If the active current cell is the first cell of the current row, the current cell will be moved to the last cell of the row previous to the current row. If the active current cell is the first cell of the first row, the focus will be moved to the previous control in the tab order of the parent container.

File: MAUI/DataGrid/selection.md [hunk 16]
```diff
@@ -0,0 +1,547 @@
...
+<td>
+Moves the current cell to the last cell of the last row.
+</td>
+</tr>
+<tr>
+<td>
+<kbd>Enter</kbd>
+</td>
+<td>
+If the active current cell is in edit mode, the changes will committed and moves the current cell to below the active current cell. If the active current cell is in last row, commits changes only and retains in the same cell.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:247]: If the active current cell is in edit mode, the changes will be committed and move the current cell below the active current cell. If the active current cell is in the last row, commits changes only remain in the same cell.

File: MAUI/DataGrid/selection.md [hunk 17]
```diff
@@ -0,0 +1,547 @@
...
+<kbd>Ctrl</kbd> + <kbd>A</kbd>
+</td>
+<td>
+All rows or cells will be selected.
+</td>
+</tr>
+</table>
+
+## Move Current Cell
+The [CurrentCell] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentCell) can be moved to a particular [RowColumnIndex] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.GridCommon.ScrollAxis.RowColumnIndex.html) by using the [SfDataGrid.MoveCurrentCellTo()] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_MoveCurrentCellTo_Syncfusion_Maui_GridCommon_ScrollAxis_RowColumnIndex_System_Boolean_) method. This method is not applicable when the `SfDataGrid.SelectionMode` is None or `NavigationMode` is Row.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:261]: The [CurrentCell] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentCell) can be moved to a particular [RowColumnIndex] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.GridCommon.ScrollAxis.RowColumnIndex.html) by using the [SfDataGrid.MoveCurrentCellTo()] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_MoveCurrentCellTo_Syncfusion_Maui_GridCommon_ScrollAxis_RowColumnIndex_System_Boolean_) method. This method is not applicable when the `SfDataGrid.SelectionMode` is None or the `NavigationMode` is Row.

File: MAUI/DataGrid/selection.md [hunk 18]
```diff
@@ -0,0 +1,547 @@
...
+        dataGrid.MoveCurrentCellTo(new RowColumnIndex(3, 1), false);
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![Move current cell in .NET MAUI DataGrid](Images/selection/maui-datagrid-movecurrentcell.png)
+
+## Clear selection
+`DataGrid` allows you to clear the selection applied in the grid rows and columns either by setting the `SfDataGrid.SelectionMode` to None or by calling the [SfDataGrid.ClearSelection()] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_ClearSelection) method.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:279]: The `DataGrid` allows you to clear the selection applied in the grid rows and columns either by setting the `SfDataGrid.SelectionMode` to None or by calling the [SfDataGrid.ClearSelection()] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_ClearSelection) method.

File: MAUI/DataGrid/selection.md [hunk 19]
```diff
@@ -0,0 +1,547 @@
...
+
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+public partial class MainPage : ContentPage
+{
+    public MainPage()
+    {
+        InitializeComponent();
+        //Clear selection using selection mode
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:288]: //Clear selection using the selection mode.

File: MAUI/DataGrid/selection.md [hunk 20]
```diff
@@ -0,0 +1,547 @@
...
+        dataGrid.SelectionMode = Syncfusion.Maui.DataGrid.DataGridSelectionMode.None;
+
+        //Clear selection using clear selection method.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:291]: //Clear selection using the clear selection method.

File: MAUI/DataGrid/selection.md [hunk 21]
```diff
@@ -0,0 +1,547 @@
...
+        dataGrid.ClearSelection();
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Events in selection
+The DataGrid provides the following events for selection:
+
+[SelectionChanging] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SelectionChanging): This event is raised while selecting a row at the execution time before the row is selected. So it allows canceling the selection action by setting the [Cancel] (https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.canceleventargs.cancel?view=net-6.0#system-componentmodel-canceleventargs-cancel) property of [DataGridSelectionChangingEventArgs] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html).
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:301]: - [SelectionChanging] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SelectionChanging): This event is raised while selecting a row at the execution time before the row is selected. So it allows canceling the selection action by setting the [Cancel] (https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.canceleventargs.cancel?view=net-6.0#system-componentmodel-canceleventargs-cancel) property of [DataGridSelectionChangingEventArgs] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html).

File: MAUI/DataGrid/selection.md [hunk 22]
```diff
@@ -0,0 +1,547 @@
...
+[SelectionChanged] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortColumnsChanged): This event is raised after a row is selected.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:302]: - [SelectionChanged] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortColumnsChanged): This event is raised after a row is selected.

File: MAUI/DataGrid/selection.md [hunk 23]
```diff
@@ -0,0 +1,547 @@
...
+These two events are triggered with `DataGridSelectionChangingEventArgs` and [DataGridSelectionChangedEventArgs] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangedEventArgs.html) that contain the following properties:
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:303]: These two events are triggered with the `DataGridSelectionChangingEventArgs` and [DataGridSelectionChangedEventArgs] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangedEventArgs.html) that contain the following properties:

File: MAUI/DataGrid/selection.md [hunk 24]
```diff
@@ -0,0 +1,547 @@
...
+
+[AddedRows] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html#Syncfusion_Maui_DataGrid_DataGridSelectionChangingEventArgs_AddedRows): Gets collection of the underlying data objects added for selection.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:305]: - [AddedRows] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html#Syncfusion_Maui_DataGrid_DataGridSelectionChangingEventArgs_AddedRows): Gets a collection of the underlying data objects added for the selection.

File: MAUI/DataGrid/selection.md [hunk 25]
```diff
@@ -0,0 +1,547 @@
...
+[RemovedRows] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html#Syncfusion_Maui_DataGrid_DataGridSelectionChangingEventArgs_RemovedRows): Gets collection of the underlying data objects removed from selection.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:306]: - [RemovedRows] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs.html#Syncfusion_Maui_DataGrid_DataGridSelectionChangingEventArgs_RemovedRows): Gets a collection of the underlying data objects removed from the selection.

File: MAUI/DataGrid/selection.md [hunk 26]
```diff
@@ -0,0 +1,547 @@
...
+
+To hook the SelectionChanging event and cancel the selection in following code example:
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:308]: To hook the SelectionChanging event and cancel the selection, follow this code example:

File: MAUI/DataGrid/selection.md [hunk 27]
```diff
@@ -0,0 +1,547 @@
...
+
+    private void DataGrid_SelectionChanging(object sender, Syncfusion.Maui.DataGrid.DataGridSelectionChangingEventArgs e)
+    {
+        e.Cancel = true;
+    }
+}
+{% enhighlight %}
+{% endtabs %}
+
+To get the selected item in code-behind using the `SelectionChanged` event, follow the code example:
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:328]: To get the selected item in a code-behind using the `SelectionChanged` event, follow this code example:

File: MAUI/DataGrid/selection.md [hunk 28]
```diff
@@ -0,0 +1,547 @@
...
+    {
+        var selectedItem = e.AddedRows[0];
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Cell Tap Events
+
+The datagrid provides the following Events to handle interactions to the cells,
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:349]: The Datagrid provides the following Events to handle interactions to the cells.

File: MAUI/DataGrid/selection.md [hunk 29]
```diff
@@ -0,0 +1,547 @@
...
+
+* [CellTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellTapped) : Called when a tap with a cell has occurred.
+* [CellDoubleTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellDoubleTapped) : Called when user is tapped a cell with a primary button at the same cell twice in quick succession.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:352]: * [CellDoubleTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellDoubleTapped): Called when the user has tapped a cell with a primary button at the same cell twice in quick succession.

File: MAUI/DataGrid/selection.md [hunk 30]
```diff
@@ -0,0 +1,547 @@
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Cell Tap Events
+
+The datagrid provides the following Events to handle interactions to the cells,
+
+* [CellTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellTapped) : Called when a tap with a cell has occurred.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:351]: * [CellTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellTapped): Called when a tap with a cell has occurred.

File: MAUI/DataGrid/selection.md [hunk 31]
```diff
@@ -0,0 +1,547 @@
...
+* [CellDoubleTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellDoubleTapped) : Called when user is tapped a cell with a primary button at the same cell twice in quick succession.
+* [CellLongPress](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellLongPress) : Called when a long press gesture with a primary button has been recognized for a cell.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:353]: * [CellLongPress](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellLongPress): Called when a long-press gesture with a primary button has been recognized for a cell.

File: MAUI/DataGrid/selection.md [hunk 32]
```diff
@@ -0,0 +1,547 @@
...
+## Cell Tap Events
+
+The datagrid provides the following Events to handle interactions to the cells,
+
+* [CellTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellTapped) : Called when a tap with a cell has occurred.
+* [CellDoubleTapped](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellDoubleTapped) : Called when user is tapped a cell with a primary button at the same cell twice in quick succession.
+* [CellLongPress](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CellLongPress) : Called when a long press gesture with a primary button has been recognized for a cell. 
+
+### CellTapped event
+This event will be triggered while tapping a cell in DataGrid. This event has [DataGridCellTappedEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellTappedEventArgs.html) as arguments.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:356]: This event will be triggered while tapping a cell in the DataGrid. This event has the [DataGridCellTappedEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellTappedEventArgs.html) as argument.

File: MAUI/DataGrid/selection.md [hunk 33]
```diff
@@ -0,0 +1,547 @@
...
+    var rowData = e.RowData;
+    var columnIndex = e.RowColumnIndex.ColumnIndex;
+    var column = e.Column;
+}
+{% endhighlight %}
+{% endtabs %}
+
+### CellDoubleTapped event
+
+This event will be triggered while double tapping a cell in DataGrid. This event has [DataGridCellDoubleTappedEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellDoubleTappedEventArgs.html) as arguments.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:377]: This event will be triggered while double-tapping a cell in the DataGrid. This event has the [DataGridCellDoubleTappedEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellDoubleTappedEventArgs.html) as argument.

File: MAUI/DataGrid/selection.md [hunk 34]
```diff
@@ -0,0 +1,547 @@
...
+    var rowData = e.RowData;
+    var columnIndex = e.RowColumnIndex.ColumnIndex;
+    var column = e.Column;
+}
+{% endhighlight %}
+{% endtabs %}
+
+### CellLongPress event
+
+This event will be triggered while long pressing a cell in DataGrid. This event has [DataGridCellLongPressEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellLongPressEventArgs.html) as arguments.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:398]: This event will be triggered while long-pressing a cell in the DataGrid. This event has the [DataGridCellLongPressEventArgs](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridCellLongPressEventArgs.html) as argument.

File: MAUI/DataGrid/selection.md [hunk 35]
```diff
@@ -0,0 +1,547 @@
...
+    var rowIndex = e.RowColumnIndex.RowIndex;
+    var rowData = e.RowData;
+    var columnIndex = e.RowColumnIndex.ColumnIndex;
+    var column = e.Column;
+}
+{% endhighlight %}
+{% endtabs %}
+
+## CurrentRow
+The [SfDataGrid.CurrentRow] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentRow) property holds the underlying data of the last selected row in data grid.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:418]: The [SfDataGrid.CurrentRow] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_CurrentRow) property holds the underlying data of the last selected row in the data grid.

File: MAUI/DataGrid/selection.md [hunk 36]
```diff
@@ -0,0 +1,547 @@
...
+
+Get the current row in the `SfDataGrid.SelectionChanged` event by setting the `SfDataGrid.SelectionMode` as Multiple or SingleDeselect. If the SelectionMode is Single, the current item and selected item are same.
+
+## 	Customizing Selection Appearance
+You can change the selection back color and text color by using [SelectionBackground] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_SelectionBackground) and [SelectedRowTextColor] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_SelectedRowTextColor) in `SfDataGrid.DefaultStyle`.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:423]: Change the selection back color and text color by using the [SelectionBackground] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_SelectionBackground) and [SelectedRowTextColor] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_SelectedRowTextColor) in the `SfDataGrid.DefaultStyle.`

File: MAUI/DataGrid/selection.md [hunk 37]
```diff
@@ -0,0 +1,547 @@
...
+        this.dataGrid.DefaultStyle.SelectionBackground = Color.FromArgb("#E3F2FD");
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![Selection style in .NET MAUI DataGrid](Images/selection/maui-datagrid-selection-style.png)
+
+## Changing current cell border color and width
+You can change the current cell border color and width by using [CurrentCellBorderColor] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_CurrentCellBorderColor) and [CurrentCellBorderWidth] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_CurrentCellBorderWidth) in SfDataGrid.DefaultStyle.
```
Comment (Chrispineagunja) [inline MAUI/DataGrid/selection.md:453]: Change the current cell border color and width by using the [CurrentCellBorderColor] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_CurrentCellBorderColor) and [CurrentCellBorderWidth] (https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridStyle.html#Syncfusion_Maui_DataGrid_DataGridStyle_CurrentCellBorderWidth) in the SfDataGrid.DefaultStyle.

**Pull request merged**